### PR TITLE
[SMALLFIX] Allows add() to return boolean

### PR DESCRIPTION
**Pull request by gpang**
Make `add()` return `boolean` like `remove()`

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java [hunk 1]
```diff
@@ -68,9 +68,10 @@ protected InodeDirectory getThis() {
    * Adds the given inode to the set of children.
    *
    * @param child the inode to add
+   * @return true if inode was added successfully, false otherwise
    */
-  public void addChild(Inode<?> child) {
-    mChildren.add(child);
+  public boolean addChild(Inode<?> child) {
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: when is the return value of this method ever used?

Comment (gpang) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: currently, it is not being used, but I want to use it in the future. Should I just keep it `void` and change it later?

Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: well, as it is, it is not clear what is the purpose of this change as nothing consumes the boolean returned by addChild (and transitively the boolean returned by add)

Comment (gpang) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: I want to be able to use it later, so this PR is first attempting to get the signature more consistent (similar to `delete`).

Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: making add / remove and addChild / removeChild symmetric sounds good
